pkg/R/RDD.R

```diff
@@ -526,6 +536,14 @@ setMethod("countByKey",
 setMethod("lapply",
           signature(X = "RDD", FUN = "function"),
           function(X, FUN) {
+            # Creats a new the closure (environment) for the FUN to capture
```

Comment:
Actually we can unify the code a little bit by implementing `lapply` using `lapplyPartitionsWithIndex`. In fact I think replacing the `PipelinedRDD` constructor here with `lapplyPartitionsWithIndex(X, func)` will do the trick.